tool/tctl/common/sessions_command.go

```diff
@@ -0,0 +1,126 @@
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gravitational/kingpin"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service"
+)
+
+// SessionsCommand implements "tctl sessions" group of commands.
+type SessionsCommand struct {
+	config *service.Config
+
+	// format is the output format (text, json, or yaml)
+	format string
+	// sessionsList implements the "tctl sessions ls" subcommand.
+	sessionsList *kingpin.CmdClause
+	// FromUTC is the start time to use for the range of sessions listed by the recorded session listing command
+	FromUTC string
+
+	// ToUTC is the start time to use for the range of sessions listed by the recorded session listing command
+	ToUTC string
+}
+
+// Initialize allows SessionsCommand to plug itself into the CLI parser
+func (c *SessionsCommand) Initialize(app *kingpin.Application, config *service.Config) {
+	c.config = config
+
+	sessions := app.Command("sessions", "Operate on recorded sessions.")
+	c.sessionsList = sessions.Command("ls", "List recorded sessions.")
+	c.sessionsList.Flag("format", "Output format, 'text', 'json', or 'yaml'").Default(teleport.Text).StringVar(&c.format)
+	c.sessionsList.Flag("from-utc", fmt.Sprintf("Start of time range in which sessions are listed. Format %s", time.RFC3339)).StringVar(&c.FromUTC)
+	c.sessionsList.Flag("to-utc", fmt.Sprintf("End of time range in which sessions are listed. Format %s", time.RFC3339)).StringVar(&c.ToUTC)
+}
+
+// TODO: Move this somewhere more appropriate
+const defaultSearchSessionPageLimit = 50
+
+// TryRun attempts to run subcommands like "sessions ls".
+func (c *SessionsCommand) TryRun(ctx context.Context, cmd string, client auth.ClientI) (match bool, err error) {
+	switch cmd {
+	case c.sessionsList.FullCommand():
+		err = c.ListSessions(client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+//TODO: Deduplicate this logic, same functions defined for tsh
+// ListApps prints the list of recorded sessions.
+func (c *SessionsCommand) ListSessions(tc auth.ClientI) error {
+	fromUTC := time.Unix(0, 0)
+	toUTC := time.Now()
+	var err error
+	if c.FromUTC != "" {
+		fromUTC, err = time.Parse(time.RFC3339, c.FromUTC)
+		if err != nil {
+			return trace.Errorf("parsing session listing start time: %v", err)
+		}
+	}
+	if c.ToUTC != "" {
+		toUTC, err = time.Parse(time.RFC3339, c.ToUTC)
+		if err != nil {
+			return trace.Errorf("parsing session listing end time: %v", err)
+		}
+	}
+
+	var sessions []events.AuditEvent
+	// Get a list of all Sessions joining pages
+	prevEventKey := ""
+	sessions = []events.AuditEvent{}
+	for {
+		nextEvents, eventKey, err := tc.SearchSessionEvents(fromUTC, toUTC,
+			defaultSearchSessionPageLimit, types.EventOrderDescending, prevEventKey, nil, "")
```

Comment:
pagination is duplicated in tsh and tctl, you could probably pull that into lib/client

Reply:
With the addition of the --limit pagination on top of that doesn't make sense to me. Now users can just specify the start time/endtime and how many they need.